lib/features/artifacts.js

```diff
@@ -128,35 +129,42 @@ export default class Artifacts {
       var putUrl =
         yield ctx.getPutUrl(taskHandler, entryName, expiry, contentType);
 
-      // Put the artifact on the server.
-      var putReq = request.put(putUrl).set({
-        'Content-Length': contentLength,
-        'Content-Type': contentType
-      });
+      let parsedUrl = url.parse(putUrl);
+      let options = {
+        hostname: parsedUrl.hostname,
+        path: parsedUrl.path,
+        method: 'PUT',
+        headers: {
+          'Content-Type': contentType,
+          'Content-Length': contentLength,
+        }
+      };
+
+      var putReq = https.request(options);
 
       // Stream tar entry to request before ending the request
       stream.pipe(putReq);
-      putReq.end();
 
-      // Wait until the response is sent.
-      var res = yield waitForEvent(putReq, 'response');
+      let response = yield new Promise((accept) => {
+        putReq.on('response', (res) => { accept(res); });
```

Comment:
What about errors? `putReq.on('error', reject);`

Note, I would fully support a method like `uploadS3Artifact(stream, putUrl, contentType, contentLength)` that returns a promise... So we can use between different files..